tensilelite/Tensile/SolutionStructs/Solution.py

```diff
@@ -2584,9 +2577,13 @@ def subCheckLdsBlockSizePerPad(tc, idx):
 
       offsetBlk = state["LdsOffsetB"] + ldsNumBytesAlignedB
 
+      mtA = state["MacroTileA"]
+      mtB = state["MacroTileB"]
+      isNonPow2MT = (mtA & (mtA - 1) != 0) or (mtB & (mtB - 1) != 0)
       state["StoreSwapAddr"] = (state["PrefetchGlobalRead"] == 2) and \
         (state["1LDSBuffer"] == 0) and \
-        (offsetBlk + int(2**(math.ceil(math.log(offsetBlk, 2)))) > depthUConfig.maxLDS)
+        (offsetBlk + int(2**(math.ceil(math.log(offsetBlk, 2)))) > depthUConfig.maxLDS) or \
+        (state["DirectToLdsA"] or state["DirectToLdsB"] and isNonPow2MT)
```

Comment:
I think you are missing () for DTLA or DTLB.
(state["DirectToLdsA"] or state["DirectToLdsB"] and isNonPow2MT)
->((state["DirectToLdsA"] or state["DirectToLdsB"]) and isNonPow2MT)

However, I think we should separately check for A and B.
For A, it does not matter if MTB is power of 2 or not.
Furthermore, it is not always true that both DTLA and DTLB are true.
(if we enable DirectToVgprA, it can be DTVA + DTLB (DTV has higher priority than DTL).

Hence, this should be like (DTLA and isNonPow2MTA) or (DTLB and isNonPow2MB).

Reply:
Good points, fixed.